Assertion 'jerry_value_is_function(jonuncaughtexception)' failed in iotjs_uncaught_exception

###### IoT.js version:

```
Checked revision: a6b2d75
Build command: tools/build.py --buildtype debug --no-snapshot --no-check-test
```
###### OS:

```
Ubuntu 17.10
```
###### Test case:

```javascript
var dns = require( 'dns' );
var assert = require( 'assert' );
assert.throws(function ( ) { dns.lookup ('localhost', require) } );
       
```

###### Backtrace:

```
uncaughtException: AssertionError: Missing expected exception..
iotjs/src/iotjs_binding_helper.c:28: Assertion 'jerry_value_is_function(jonuncaughtexception)' failed.

(gdb) bt
#0  iotjs_uncaught_exception (jexception=28306) at iotjs/src/iotjs_binding_helper.c:28
#1  0x000055555556f1fa in iotjs_make_callback_with_result (jfunction=23739, jthis=72, jargs=0x7fffffffa660)
    at iotjs/src/iotjs_binding_helper.c:121
#2  0x000055555556f19f in iotjs_make_callback (jfunction=23739, jthis=72, jargs=0x7fffffffa660)
    at iotjs/src/iotjs_binding_helper.c:108
#3  0x0000555555575402 in AfterGetAddrInfo (req=0x55555588c830, status=0, res=0x7ffff0001be0)
    at iotjs/src/modules/iotjs_module_dns.c:158
#4  0x00005555555f3198 in uv__getaddrinfo_done (w=0x55555588c858, status=0)
    at iotjs/deps/libtuv/src/unix/getaddrinfo.c:156
#5  0x00005555555fcb88 in uv__work_done (handle=0x55555584a6f0 <default_loop_struct+176>)
    at iotjs/deps/libtuv/src/threadpool.c:261
#6  0x00005555555ee2fc in uv__async_event (loop=0x55555584a640 <default_loop_struct>, w=0x55555584a7e8 <default_loop_struct+424>, 
    nevents=1) at iotjs/deps/libtuv/src/unix/async.c:122
#7  0x00005555555ee475 in uv__async_io (loop=0x55555584a640 <default_loop_struct>, w=0x55555584a7f0 <default_loop_struct+432>, events=1)
    at iotjs/deps/libtuv/src/unix/async.c:162
#8  0x00005555555fbdeb in uv__io_poll (loop=0x55555584a640 <default_loop_struct>, timeout=-1)
    at iotjs/deps/libtuv/src/unix/linux-core.c:389
#9  0x00005555555eeeaa in uv_run (loop=0x55555584a640 <default_loop_struct>, mode=UV_RUN_DEFAULT)
    at iotjs/deps/libtuv/src/unix/core.c:305
#10 0x000055555556cd24 in iotjs_entry (argc=2, argv=0x7fffffffdec8) at iotjs/src/iotjs.c:222
#11 0x000055555556c4ca in main (argc=2, argv=0x7fffffffdec8) at iotjs/iotjs_linux.c:19
```

<sup>Found by [Fuzzinator](http://fuzzinator.readthedocs.io/) with [grammarinator](https://github.com/renatahodovan/grammarinator)</sup>

--- a/src/iotjs_binding_helper.c
+++ b/src/iotjs_binding_helper.c
@@ -116,6 +116,10 @@ void iotjs_make_callback(jerry_value_t jfunction, jerry_value_t jthis,
 jerry_value_t iotjs_make_callback_with_result(jerry_value_t jfunction,
                                               jerry_value_t jthis,
                                               const iotjs_jargs_t* jargs) {
+  // If the environment is already exiting just return an undefined value.
+  if (iotjs_environment_is_exiting(iotjs_environment_get())) {
+    return jerry_create_undefined();
+  }
   // Calls back the function.
   jerry_value_t jres = iotjs_jhelper_call(jfunction, jthis, jargs);
   if (jerry_value_has_error_flag(jres)) {

--- a/test/run_fail/test-issue-1349.js
+++ b/test/run_fail/test-issue-1349.js
@@ -1,0 +1,18 @@
+/* Copyright 2017-present Samsung Electronics Co., Ltd. and other contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var dns = require( 'dns' );
+var assert = require( 'assert' );
+assert.throws(function ( ) { dns.lookup ('localhost', require) } );

--- a/test/testsets.json
+++ b/test/testsets.json
@@ -123,6 +123,7 @@
     { "name": "test_fs_callbacks_called.js", "expected-failure": true },
     { "name": "test_iotjs_runtime_error.js", "expected-failure": true },
     { "name": "test_iotjs_syntax_error.js", "expected-failure": true },
+    { "name": "test-issue-1349.js", "expected-failure": true},
     { "name": "test-issue-1360.js", "expected-failure": true},
     { "name": "test_module_require_invalid_file.js", "expected-failure": true },
     { "name": "test_module_require_path_below_root.js", "expected-failure": true },

